### PR TITLE
loadbalancingexporter: add support for logs - 4/5 + 5/5

### LINKED ISSUE #2470: loadbalancing exporter: Add support for logs
**Is your feature request related to a problem? Please describe.**
Allow logs to be exported based on the same logic as traces (using traceID hashing) to enable tail sampling for distributed logs for requests spanning multiple services.

**Describe the solution you'd like**
Add implementation for the `component.LogExporter` interface to the existing `loadbalancingexporter`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by BinaryHexer**
**Description:** <Describe what has changed.>

Exporter activation in the factory.

- In `exporter/loadbalancingexporter`
  - Add `createLogExporter` and use it in `NewFactory` in `factory.go`.
  - Add tests in `factory_test.go` and rename some existing tests.

**Link to tracking Issue:** 
- next step after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2699
- closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2470

Testing: Unit tests

Documentation: README.md

Signed-off-by: Vijayender Joshi vijayeder.joshi@gmail.com

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
I'm really happy with how it turned out, @BinaryHexer! Thanks for this feature. Before approving, I just wanted to make sure you did a manual test before with this code. Would you mind providing the steps used in the test?

**Review by jpkrohling — APPROVED** [review]

**Comment by BinaryHexer** [comment]
I have added an example directory that contains a basic docker setup for an easy reproducible demo the loadbalancing for logging.

I have done this as there are some other exporters that also contain such an example directory, if this is not acceptable, I will remove it, and can just mention those same steps just as part of this PR.

[`6f36b3d` (#2721)](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2721/commits/6f36b3d992b208690af1181f9c2be1b420da283e)

**Comment by BinaryHexer** [comment]
Thank you @jpkrohling for your reviews and helping me complete this feature! 🙇‍♂️ 

Also appreciate @bogdandrutu for merging all the PRs related to this feature 😄

**Review by bogdandrutu — APPROVED** [review]

File: exporter/loadbalancingexporter/testdata/config.yaml [hunk 1]
```diff
@@ -42,3 +42,9 @@ service:
       processors: []
       exporters:
       - loadbalancing
+    logs:
+      receivers:
+        - nop
+      processors: []
+      exporters:
+        - loadbalancing
```
[the remark this answers is not in the document]
Comment (BinaryHexer) [reply exporter/loadbalancingexporter/testdata/config.yaml]: Fixed in [`bbe23a0` (#2721)](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2721/commits/bbe23a098036b20cd646cf2871d83b54834ab97c)

**Pull request merged**